WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -436,7 +438,10 @@ private async Task OnConfirmAsync()
 
 			_info.UserLabels = result.Result;
 
-			var transaction = await Task.Run(() => TransactionHelpers.BuildTransaction(_wallet, _info));
+			var transaction = _info.IsOptimized ?
+				await Task.Run(() => TransactionHelpers.BuildChangelessTransaction(_wallet, _info.Address, _info.UserLabels, _info.FeeRate, _info.ChangelessCoins))
+				: await Task.Run(() => TransactionHelpers.BuildTransaction(_wallet, _info));
```

Comment:
What is the purpose of `Task.Run` here given that those `TransactionHelpers` methods are synchronous?

Reply:
We use `Task.Run` in every `ViewModel`, where we need to build up a TX. I would rather not change that.

Reply:
Yeah, but why?